### PR TITLE
Tone merkle path trim

### PR DESCRIPTION
**Pull request by tonesnotes**
Added trim() method to MerklePath which removes all computable leaves.

Added call to the new trim() method at the end of combine() so MerklePath size is minimized.

### CODE REVIEW

**Review by ty-everett — APPROVED** [review]
Looks good to me, I'd like to page @sirdeggen just to make sure this aligns with the requirements from the Go side, and with his approval, we'll get it merged.

**Comment by sirdeggen** [comment]
Thanks gents taking a close look now.

**Comment by sirdeggen** [comment]
Something seems not quite right. I think when you feed it a BUMP with only the bottom level defined it doesn't realize there are supposed to be levels above it. The tree height is not set by the number of path elements on the bottom, but the number of arrays or something like that.

File: src/transaction/MerklePath.ts [hunk 1]
```diff
@@ -194,7 +201,7 @@ export default class MerklePath {
     for (let height = 0; height < this.path.length; height++) {
```
Comment (sirdeggen) [inline src/transaction/MerklePath.ts:201]: this.path.length when using trimmed object will not be the same as tree height.

Comment (sirdeggen) [reply src/transaction/MerklePath.ts:202]: empty arrays need to be set, which was previously not allowed but now is.

**Review by sirdeggen — APPROVED** [review]

**Pull request merged**